### PR TITLE
Backend performance optimizations

### LINKED ISSUE #151: Backend performance pass
We've been building out each section while ignoring Django query performance additions such as `select_related` and `prefetch_related`.  This hasn't been an issue with the small amounts of local data we've been working with but we'll want these in place before production.

 - [ ] Add select_related and prefetch_related to our backend objects.

### PR DESCRIPTION
**Pull request by rmeyers4**
Closes #151

### CODE REVIEW

**Comment by rmeyers4** [comment]
@trevorskaggs : Ready for review!

File: shelter/serializers.py [hunk 1]
```diff
@@ -8,14 +8,14 @@
 from animals.serializers import AnimalSerializer
 
 class RoomSerializer(serializers.ModelSerializer):
-    animals = AnimalSerializer(source='animal_set', many=True, required=False, read_only=True)
+    # animals = AnimalSerializer(source='animal_set', many=True, required=False, read_only=True)
```
Comment (AlexMountain) [inline shelter/serializers.py:11]: We need this one added back in some capacity as well.

File: hotline/serializers.py [hunk 2]
```diff
@@ -78,14 +94,35 @@ class Meta:
...
+        if self.context.get('request') and self.context.get('request').path == '/hotline/api/servicerequests/':
+            self.fields.pop('action_history')
+            self.fields.pop('assigned_evac')
+
+        # if fields is not None:
+        #     # Drop any fields that are not specified in the `fields` argument.
+        #     allowed = set(fields)
+        #     existing = set(self.fields)
+        #     for field_name in existing - allowed:
+        #         self.fields.pop(field_name)
```
Comment (AlexMountain) [inline hotline/serializers.py:128]: Are we saving this for later?

**Review by AlexMountain — COMMENTED** [review]
Sweet sweet speedy action.  We seem to be missing a couple of returns however causing some pages to explode.  With those added back I think we'll be good!

File: animals/serializers.py [hunk 3]
```diff
@@ -5,27 +5,16 @@
...
     request_address = serializers.SerializerMethodField()
     aco_required = serializers.SerializerMethodField()
     front_image = serializers.SerializerMethodField()
     side_image = serializers.SerializerMethodField()
     extra_images = serializers.SerializerMethodField()
-    action_history = serializers.SerializerMethodField()
     shelter_name = serializers.SerializerMethodField()
     shelter = serializers.SerializerMethodField()
-    is_stray = serializers.SerializerMethodField()
-    evacuation_assignments = serializers.SerializerMethodField()
```
[the remark this answers is not in the document]
Comment (rmeyers4) [reply animals/serializers.py:20]: Good call, I added it back to the `AnimalSerializer` and added a prefetch.

File: shelter/serializers.py (hunk 1 above)
Comment (rmeyers4) [reply shelter/serializers.py:11]: Nice catch, I got overzealous at the beginning and meant to get to this one. Will restore and address shelter performance in a different ticket.

File: hotline/serializers.py (hunk 2 above)
Comment (rmeyers4) [reply hotline/serializers.py]: Ah, this was copy/pasted code from a DRF example, not needed!

**Comment by rmeyers4** [comment]
@AlexMountain : Thanks for the thorough review! I think I addressed the issues you identified. I clicked through the Dispatch process and nothing exploded. I think some additional optimization will be needed for Dispatch and Shelter but those can be addressed in a different ticket.

**Review by trevorskaggs — APPROVED** [review]
SNAP, these are awesome.  Looking forward to seeing in production!

**Review by AlexMountain — APPROVED** [review]
Awesome dude, good to go!

**Pull request merged**